custom/git-app-deployer/main.go

```diff
@@ -0,0 +1,183 @@
+package main
+
+import (
+	"bytes"
+	"flag"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/litmuschaos/test-tools/pkg/log"
+	"github.com/openebs/maya/pkg/util/retry"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/homedir"
+)
+
+func main() {
+
+	var filePath string
+	var timeout int
+	namespace := "sock-shop"
+
+	// filePath is path for manifest, namespace is of application, timeout for application
+	filePath, timeout = GetData()
+
+	//Get kubeconfig
+	config, err := GetKubeConfig()
+	if err != nil {
+		panic(err.Error())
+	}
+
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	log.Info("[Status]: App-deployer is started...")
+	log.Info(`[Status]: App-deployer type is: ` + filePath + `-sock-shop.`)
+
+	if err := CreateNamespace(clientset, namespace); err != nil {
+		log.Fatalf("Failed to create the namespace due to, err: %v", err)
+	}
+	log.Info("[Status]: Namespace shock-shop is successfully created!")
+
+	if err := CreateShockShop(`/var/run/`+filePath+`-sock-shop.yaml`, namespace); err != nil {
+		log.Fatalf("Error with creating experiment: %v", err)
+	}
+	log.Info("[Status]: sock-shop applications is successfully created!")
+
+	if err := CheckApplicationStatus(namespace, "app=sock-shop", timeout, 2, clientset); err != nil {
+		log.Fatalf("err: %v", err)
+	}
+
+}
+
+// GetKubeConfig function... If In-Cluster is nil then it will go for Out-Cluster config
+func GetKubeConfig() (*rest.Config, error) {
+	var kubeconfig *string
+	config, err := rest.InClusterConfig()
+	if err != nil {
+		//To get Out-Cluster config
+		if home := homedir.HomeDir(); home != "" {
+			kubeconfig = flag.String("kubeconfig", filepath.Join(home, ".kube", "config"), "kubeconfig file it is out-of-cluster")
+		} else {
+			kubeconfig = flag.String("kubeconfig", "", "Path to the kubeconfig file")
+		}
+		flag.Parse()
+		config, err = clientcmd.BuildConfigFromFlags("", *kubeconfig)
+	}
+	return config, err
+}
+
+//GetData is used for type of Scenarios and timeout.
+func GetData() (string, int) {
+	timeout := 200
+	typeName := flag.String("typeName", "", "absolute type of the sock-shop")
+	flag.Parse()
+
+	if *typeName == "" {
+		*typeName = "week"
```

Comment:
it's week or weak? and please check the name of the manifest as well. `week-sock-shop.yaml`